catalog/core/catalog-email-impl/src/main/java/org/codice/alliance/core/email/impl/EmailSenderImpl.java

```diff
@@ -0,0 +1,188 @@
+import org.codice.alliance.core.email.EmailSender;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EmailSenderImpl implements EmailSender {
+
+    private static final int BUFFER_SIZE = 4096;
+
+    private static final int DEFAULT_PORT = 25;
+
+    private static final long DEFAULT_MAX_ATTACHMENT_SIZE = Long.MAX_VALUE;
+
+    private static final String SMTP_HOST_PROPERTY = "mail.smtp.host";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EmailSenderImpl.class);
+
+    private static final String SMTP_PORT_PROPERTY = "mail.smtp.port";
+
+    private static final String SMTP_AUTH_PROPERTY = "mail.smtp.auth";
+
+    private int mailPort = DEFAULT_PORT;
+
+    private String mailHost;
+
+    private long maxAttachmentSize = DEFAULT_MAX_ATTACHMENT_SIZE;
+
+    public void setMailPort(int mailPort) {
+        this.mailPort = mailPort;
+    }
+
+    public void setMailHost(String mailHost) {
+        this.mailHost = mailHost;
+    }
+
+    public void setMaxAttachmentSize(long maxAttachmentSize) {
+        this.maxAttachmentSize = maxAttachmentSize;
+    }
+
+    /**
+     * sendEmail method sends email after receiving input parameters
+     */
+    @Override
+    public void sendEmail(String fromEmail, String toEmail, String subject, String body,
+            List<Pair<String, InputStream>> attachments) throws IOException {
+        notNull(fromEmail, "fromEmail must be non-null");
+        notNull(toEmail, "toEmail must be non-null");
+        notNull(subject, "subject must be non-null");
+        notNull(body, "body must be non-null");
+        notNull(attachments, "attachments must be non-null");
+
+        if (StringUtils.isBlank(mailHost)) {
+            throw new IOException("the mail server hostname has not been configured");
+        }
+
+        List<File> tempFiles = new LinkedList<>();
+
+        try {
+            InternetAddress emailAddr = new InternetAddress(toEmail);
+            emailAddr.validate();
+
+            Properties properties = createSessionProperties();
+
+            Session session = Session.getDefaultInstance(properties);
+
+            MimeMessage mimeMessage = new MimeMessage(session);
+            mimeMessage.setFrom(new InternetAddress(fromEmail));
+            mimeMessage.addRecipient(Message.RecipientType.TO, emailAddr);
+            mimeMessage.setSubject(subject);
+
+            BodyPart messageBodyPart = new MimeBodyPart();
+            messageBodyPart.setText(body);
+
+            Multipart multipart = new MimeMultipart();
+            multipart.addBodyPart(messageBodyPart);
+
+            Holder<Long> bytesWritten = new Holder<>(0L);
+
+            for (Pair<String, InputStream> attachment : attachments) {
+
+                messageBodyPart = new MimeBodyPart();
+                File file = File.createTempFile("email-sender-", ".dat");
+                tempFiles.add(file);
+
+                copyDataToTempFile(file, attachment.getValue(), bytesWritten);
```

Comment:
The generalized email sender supports multiple attachments, but the order request code calls the email sender with only one attachment at a time. I'm not sure if the email sender code should try to split up multiple attachments into separate emails, at least not without a flag the caller can use to indicate that it is OK. The caller may want all attachments to be sent or none at all.